File: _benchmarks/iris-mvc-templates/controllers/home_controller.go
package controllers

import "github.com/kataras/iris/mvc"

type HomeController struct{}

func (c *HomeController) Get() mvc.Result {
	return mvc.View{Name: "index.html"}
}

func (c *HomeController) GetAbout() mvc.Result {
	return mvc.View{
		Name: "about.html",
		Data: map[string]interface{}{
			"Title":   "About Page",
			"Message": "Your application description page."},
	}
}

func (c *HomeController) GetContact() mvc.Result {
	return mvc.View{
		Name: "contact.html",
		Data: map[string]interface{}{
			"Title":   "Contact Page",
			"Message": "Your application description page."},
	}
}

File: _benchmarks/netcore-mvc-templates/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using netcore_mvc_templates.Models;

namespace netcore_mvc_templates.Controllers
{
    public class HomeController : Controller
    {
        public IActionResult Index()
        {
            return View();
        }

        public IActionResult About()
        {
            ViewData["Title"] = "About Page";
            ViewData["Message"] = "Your application description page.";

            return View();
        }

        public IActionResult Contact()
        {
            ViewData["Title"] = "Contact Page";
            ViewData["Message"] = "Your contact page.";

            return View();
        }

        public IActionResult Error()
        {
            return View(new ErrorViewModel { Title = "Error", Code = 500});
        }
    }
}
